File: Qb.Js/src/Constant.ts
import { OmConstant } from "./OmConstant";

/**
 * Класс описания константы
 */
export class Constant {
  private Const: OmConstant;

  static Date(val: Date): Constant {
    var res = new Constant();
    res.Const = OmConstant.Date(val);
    return res;
  }

  /**
   * Создает экземтляр класса Constant со строковым значением
   * @param  val - строковая константа
   */
  static String(val: string): Constant {
    var res = new Constant();
    res.Const = OmConstant.String(val);
    return res;
  }

  /**
   * Создает экземтляр класса Constant с числовым значением
   * @param val - числовая константа
   */
  static Number(val: number): Constant {
    var res = new Constant();
    res.Const = OmConstant.Number(val);
    return res;
  }

}

File: Qb.Net/Constant.cs
using Viten.QueryBuilder.SqlOm;
using System;

namespace Viten.QueryBuilder
{

  /// <summary>Класс описания константы</summary>
  public class Constant
  {
    internal OmConstant Const;
    private Constant()
    {
    }

    /// <summary>Константа</summary>
    public static Constant Date(DateTime val)
    {
      Constant retVal = new Constant();
      retVal.Const = OmConstant.Date(val);
      return retVal;
    }

    /// <summary>Константа</summary>
    public static Constant String(string val)
    {
      Constant retVal = new Constant();
      retVal.Const = OmConstant.String(val);
      return retVal;
    }

    /// <summary>Константа</summary>
    public static Constant Num(int val)
    {
      Constant retVal = new Constant();
      retVal.Const = OmConstant.Number(val);
      return retVal;
    }

    /// <summary>Константа</summary>
    public static Constant Num(long val)
    {
      Constant retVal = new Constant();
      retVal.Const = OmConstant.Number(val);
      return retVal;
    }

    /// <summary>Константа</summary>
    public static Constant Num(double val)
    {
      Constant retVal = new Constant();
      retVal.Const = OmConstant.Number(val);
      return retVal;
    }
  }
}
